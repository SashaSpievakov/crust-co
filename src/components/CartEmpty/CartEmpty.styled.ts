/* eslint import/prefer-default-export: 0 */

import styled from 'styled-components';

const Section = styled.section`
  margin: 80px auto;
  max-width: 600px;
  text-align: center;

  h2 {
    font-size: 36px;
    font-weight: 700;
    line-height: 1.4;
    margin-bottom: 25px;

    @media screen and (max-width: 500px) {
      font-size: 30px;
    }
  }

  p {
    font-size: 20px;
    line-height: 145.4%;
    letter-spacing: 0.01em;
    color: #777777;

    @media screen and (max-width: 500px) {
      font-size: 15px;
    }
  }

  img {
    display: block;
    width: 300px;
    margin: 45px auto 60px;

    @media screen and (max-width: 500px) {
      width: 200px;
      height: 170px;
    }
  }
`;

export { Section };
