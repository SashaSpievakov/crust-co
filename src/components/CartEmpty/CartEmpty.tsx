import { Link } from 'react-router-dom';

import cartImg from '../../assets/img/empty-cart.png';
import { ButtonBlack } from '../../styles/Buttons.styled';
import Article from './CartEmpty.styled';

const CartEmpty = () => {
  return (
    <Article>
      <h2>
        Your cart is empty <span>ðŸ˜•</span>
      </h2>
      <p>
        You haven&apos;t ordered food yet.
        <br />
        If you want to make an order, go back to the main page.
      </p>
      <img src={cartImg} alt="empty cart" width="300" height="255" />
      <ButtonBlack as={Link} to="/">
        <span>Go Back</span>
      </ButtonBlack>
    </Article>
  );
};

export default CartEmpty;
