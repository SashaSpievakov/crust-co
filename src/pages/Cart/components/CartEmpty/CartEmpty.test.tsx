import '@testing-library/jest-dom';

import { rendererWithProviders } from '../../../../tests/helpers';
import { CartEmpty } from './CartEmpty';

describe('CartEmpty tests', () => {
  test('should render the CartEmpty component', () => {
    const snapshot = rendererWithProviders(<CartEmpty />);
    expect(snapshot).toMatchSnapshot();
  });
});
