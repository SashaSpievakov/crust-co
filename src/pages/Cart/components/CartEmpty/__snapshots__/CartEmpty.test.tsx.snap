// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartEmpty Tests renders the CartEmpty component 1`] = `
Object {
  "children": Array [
    <h2>
      Your cart is empty 
      <span>
        ðŸ˜•
      </span>
    </h2>,
    <p>
      You haven't ordered food yet.
      <br />
      If you want to make an order, go back to the main page.
    </p>,
    <img
      alt="empty cart"
      height="255"
      src="empty-cart.png"
      width="300"
    />,
    <a
      className="sc-bcXHqe sc-gswNZR glVyWT bMYeao"
      href="/"
      onClick={[Function]}
    >
      Go Back
    </a>,
  ],
  "props": Object {
    "className": "sc-lbVpMG enMBOd",
  },
  "type": "article",
}
`;
